File: src/minigames/treasure_hunter/java/work/lclpnet/ap2/game/treasure_hunter/TreasureHunterMinigame.java
package work.lclpnet.ap2.game.treasure_hunter;

import net.minecraft.core.RegistryAccess;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import org.jetbrains.annotations.NotNull;
import work.lclpnet.ap2.ApConstants;
import work.lclpnet.ap2.api.game.*;

public class TreasureHunterMinigame implements MiniGame {
    @Override
    public @NotNull ResourceLocation getId() {
        return ApConstants.identifier("treasure_hunter");
    }

    @Override
    public @NotNull GameType getType() {
        return GameType.FFA;
    }

    @Override
    public @NotNull String getAuthor() {
        return ApConstants.PERSON_BOPS;
    }

    @Override
    public @NotNull ItemStack getIcon(@NotNull RegistryAccess manager) {
        return new ItemStack(Items.CHEST);
    }

    @Override
    public boolean canBeFinale(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public @NotNull MiniGameInstance createInstance(@NotNull MiniGameHandle gameHandle) {
        return new TreasureHunterInstance(gameHandle);
    }
}

File: src/minigames/treasure_hunter/src/main/kotlin/work/lclpnet/ap2/game/treasure_hunter/TreasureHunterMinigame.kt
package work.lclpnet.ap2.game.treasure_hunter

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.base.GameStartContext
import work.lclpnet.ap2.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.util.MapLevelGameFactory

class TreasureHunterMinigame : MiniGame {
    override val id = ApConstants.identifier("treasure_hunter")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_BOPS
    override fun getIcon(manager: RegistryAccess) = ItemStack(Items.CHEST)
    override fun canBeFinale(context: GameStartContext) = true
    override fun canBePlayed(context: GameStartContext) = true
    override fun createFactory() = MapLevelGameFactory(::TreasureHunterInstance)
}
